File: src/main/java/rest/Command.java
package rest;

public class Command {
    private int id;
    private int subsystem_id;
    private String command;
    private String result;
    private String timestamp;

    public Command(int id, int subsystem_id, String command, String result, String timestamp) {
        this.id = id;
        this.subsystem_id = subsystem_id;
        this.command = command;
        this.result=result;
        this.timestamp=timestamp;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id=id;
    }

    public int getSubsystem_id() {
        return subsystem_id;
    }

    public void setSubsystem_id(int subsystem_id) {
        this.subsystem_id = subsystem_id;
    }

    public String getCommand() {
        return command;
    }

    public void setCommand(String command) {
        this.command = command;
    }

    public String getResult() {
        return result;
    }

    public void setResult(String result) {
        this.result = result;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(String timestamp) {
        this.timestamp = timestamp;
    }
}

File: src/app/data/Command.ts
export class Command {

  private static idCounter=0;
  private id: number;
  private subsystemId: number;
  private command: string;
  private result: string;
  private timestamp: string;

  constructor(subsystemId, command) {
    const date = new Date();
    Command.idCounter++;
    this.id=Command.idCounter;
    this.subsystemId=subsystemId;
    this.command=command;
    this.timestamp=date.getDate()+'-'+date.getMonth()+'-'+date.getFullYear()+' '
      +date.getHours()+':'+date.getMinutes()+':'+date.getSeconds();
    this.result='';
  }

  getId(): number {
    return this.id;
  }

  serId(value: number) {
    this.id = value;
  }

  getSubsystemId(): number {
    return this.subsystemId;
  }

  setSubsystemId(value: number) {
    this.subsystemId = value;
  }

  getCommand(): string {
    return this.command;
  }

  setCommand(value: string) {
    this.command = value;
  }

  getResult(): string {
    return this.result;
  }

  setResult(value: string) {
    this.result = value;
  }

  getTimestamp(): string {
    return this.timestamp;
  }

  setTimestamp(value: string) {
    this.timestamp = value;
  }
}
